### ISSUE TITLE
Potential crash in lua interface when nil is passed to some functions.

### ISSUE THREAD

**Issue by slavidodo** [otland/forgottenserver#2695]
### Before creating an issue, please ensure:
- [x] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [x] This issue is reproducible without changes to the code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. call monster:selectTarget(nil)

### Expected behaviour
return false

### Actual behaviour
segmentation fault:
read access violation **creature** was nullptr.

## More information
There are a lot of functions in lua that don't check if the passed param is nullptr.

**Comment by Lyuzera** [comment]
try adding this check to the method  `bool Monster::selectTarget(Creature* creature)` file `monster.cpp`
```C
	if (!creature) {
		return false;
	}
```

**Comment by slavidodo** [comment]
These functions should accept references instead so you're sure that you're not passing nullptr.

I'm already working on a solution~

**Comment by Lyuzera** [comment]
I believe that if the locking happens with LUA scripts, you should be careful when writing the code, because to avoid this, it is necessary to check if the creature exists.
```Lua
if creature then
	monster:selectTarget(creature)
end
```
In urgent cases, you can add this temporarily in the `Monster::isTarget` method:
```C
	if (!creature) {
		std::cout << "[Warning - Monster::isTarget] creature doesn't exist." << std::endl;
		return false;
	}
```
but I agree with you, it's not cool to pass nullptr because otherwise the creature never selects a target.

**Comment by Zbizu** [comment]
A crash is still a crash and should be fixed when it's easy to fix.

**Comment by marmichalski** [comment]
This has been resolved by @MillhioreBT in #3562.

**Comment by EPuncker** [comment]
closed with https://github.com/otland/forgottenserver/commit/de27b23e0efb245b88d188888981b866e58f8edd
